src/items.rs

```diff
@@ -790,6 +759,37 @@ impl<'a> FmtVisitor<'a> {
     }
 }
 
+fn rewrite_explicit_self(explicit_self: &ast::ExplicitSelf, args: &[ast::Arg]) -> String {
```

Comment:
Better to return `Option<String>` than giving a special meaning to the empty string (also saves an allocation).